### PR TITLE
Fixes a blockage occurring with hotbackup when writing.

### PR DESCRIPTION
**Pull request by neunhoef**
This is a forward port of a fix for 3.6.
The problem is that hotbackup tries to prevent write transactions and then
wants to wait until all write transactions on all dbservers have finished.
However, when a write transaction on a leader has already started, the
corresponding follower transaction for the replication might no longer
be able to start, resulting in a blockage.

This PR fixes this by only stopping leader transactions.

### CODE REVIEW

File: arangod/ClusterEngine/ClusterTransactionState.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ Result ClusterTransactionState::beginTransaction(transaction::Hints hints) {
   updateStatus(transaction::Status::RUNNING);
   _vocbase.server().getFeature<MetricsFeature>().serverStatistics()._transactionsStatistics._transactionsStarted++;
 
-  transaction::ManagerFeature::manager()->registerTransaction(id(), isReadOnlyTransaction());
+  transaction::ManagerFeature::manager()->registerTransaction(id(), isReadOnlyTransaction(), hasHint(transaction::Hints::Hint::IS_FOLLOWER_TRX));
```
Comment (graetzer) [inline arangod/ClusterEngine/ClusterTransactionState.cpp:83]: why would this be supported on the coordinator

File: arangod/RestHandler/RestDocumentHandler.cpp [hunk 2]
```diff
@@ -203,6 +203,9 @@ RestStatus RestDocumentHandler::insertDocument() {
   if (!isMultiple && !opOptions.isOverwriteModeUpdateReplace()) {
     _activeTrx->addHint(transaction::Hints::Hint::SINGLE_OPERATION);
   }
+  if (!opOptions.isSynchronousReplicationFrom.empty()) {
```
Comment (graetzer) [inline arangod/RestHandler/RestDocumentHandler.cpp:206]: this might be not empty on a single-server as well no?

Comment (neunhoef) [reply arangod/RestHandler/RestDocumentHandler.cpp:206]: I think it is only non-empty if this is a follower in a cluster. If it is non-empty on a single server then there are no further consequences, even if the hint is set for the transaction.

File: arangod/ClusterEngine/ClusterTransactionState.cpp (hunk 1 above)
Comment (neunhoef) [reply arangod/ClusterEngine/ClusterTransactionState.cpp:83]: Just for uniformity. On a coordinator no transaction will ever be a "followertransaction". We might also just put `false` here, if you prefer.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11416/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11418/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11425/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11428/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11435/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11458/

File: arangod/ClusterEngine/ClusterTransactionState.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/ClusterEngine/ClusterTransactionState.cpp:83]: I also think we should hard-code this to `false`.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11469/

File: arangod/ClusterEngine/ClusterTransactionState.cpp (hunk 1 above)
Comment (neunhoef) [reply arangod/ClusterEngine/ClusterTransactionState.cpp]: Agreed, did the change. But I think this is only necessary for `devel`, not for the already merged PRs in the other branches.

**Pull request merged**